base/random.jl

```diff
@@ -36,24 +40,25 @@ if is_windows()
         RandomDevice() = new(Vector{UInt128}(1))
     end
 
-    function rand(rd::RandomDevice, ::Type{T}) where T<:Union{Bool,Base.BitInteger}
+    function rand(rd::RandomDevice, T::BoolBitIntegerType)
         win32_SystemFunction036!(rd.buffer)
         @inbounds return rd.buffer[1] % T
     end
 
-    rand!(rd::RandomDevice, A::Array{<:Union{Bool,Base.BitInteger}}) = (win32_SystemFunction036!(A); A)
+    rand!(rd::RandomDevice, A::BoolBitIntegerArray) = (win32_SystemFunction036!(A); A)
+
 else # !windows
+
```

Comment:
I just meant it's rare to have struct or method definition within an if block, it's usually at the top level; sometimes the style is amended for special cases, like skipping the indentation corresponding module blocks, I thought why not here, but no pb!